packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -185,6 +187,250 @@ class WebpackCLI {
+
+                    if (isPromise) {
+                        rawConfig = await rawConfig;
+                    }
+
+                    // `Promise` may return `Function`
+                    if (typeof rawConfig === 'function') {
+                        // when config is a function, pass the env from args to the config function
+                        rawConfig = await rawConfig(env, args);
+                    }
+
+                    return rawConfig;
+                }),
+            );
+
+            for (const singleConfig of allConfigs) {
+                if (Array.isArray(singleConfig)) {
+                    configs.push(...singleConfig);
+                } else {
+                    configs.push(singleConfig);
+                }
+            }
+
+            if (configName) {
+                const foundConfigNames = [];
+
+                configs = configs.filter((options) => {
+                    const found = configName.includes(options.name);
+
+                    if (found) {
+                        foundConfigNames.push(options.name);
+                    }
+
+                    return found;
+                });
+
+                if (foundConfigNames.length !== configName.length) {
+                    // Configuration with name "test" was not found.
+                    logger.error(
+                        configName
+                            .filter((name) => !foundConfigNames.includes(name))
+                            .map((configName) => `Configuration with name "${configName}" was not found.`)
+                            .join('\n'),
+                    );
+                    process.exit(2);
+                }
+            }
+
+            if (configs.length === 0) {
+                logger.error('No configurations found');
+                process.exit(2);
+            }
+
+            newOptionsObject['options'] = configs.length > 1 ? configs : configs[0];
+
+            return newOptionsObject;
+        };
+
+        const resolveConfigMerging = async (args) => {
+            const { merge } = args;
+
+            if (merge) {
+                // Get the current configuration options
+                const { options: configOptions } = opts;
+
+                // we can only merge when there are multiple configurations
+                // either by passing multiple configs by flags or passing a
+                // single config exporting an array
+                if (!Array.isArray(configOptions)) {
+                    logger.error('At least two configurations are required for merge.');
+                    process.exit(2);
+                }
+
+                // We return a single config object which is passed to the compiler
+                opts['options'] = configOptions.reduce((currentConfig, mergedConfig) => webpackMerge(currentConfig, mergedConfig), {});
+            }
+        };
+        await resolveConfigFiles(args);
+        await resolveConfigMerging(args);
```

Comment:
Inline it, no need extra function and then run it, because we use it only one

Reply:
agree, done